tt-train/sources/ttml/CMakeLists.txt

```diff
@@ -0,0 +1,111 @@
+project(ttml)
+
+file(
+    GLOB_RECURSE SOURCES
+    LIST_DIRECTORIES true
+    *.hpp
+    *.cpp
+) # I am a bad person
+
+add_library(ttml STATIC ${SOURCES})
+
+target_include_directories(
+    ttml
+    SYSTEM
+    PUBLIC
```

Comment:
Is anything here actually `PUBLIC`?  Does `ttml` even have a public API that leaks headers from Metal and TTNN?